File: typescipt/src/parser/IParserLogger.ts
import {SourceReference} from "./sourceReference";
import {IRootNode} from "../language/iRootNode";

export interface IParserLogger {

  logInfo(message: string): void;
  log(reference: SourceReference, message: string): void;
  fail(reference: SourceReference, message: string): void;

  /*
  fail(node: INode, reference: SourceReference, message: string);

  logNodes(IEnumerable<INode> nodes);

*/

  hasErrors(): boolean;

  /*
  bool HasRootErrors();

*/

  hasErrorMessage(expectedError: string): boolean;

  formatMessages(): string;

  /*
    bool NodeHasErrors(IRootNode node);

    string[] ErrorMessages();
    string[] ErrorRootMessages();
    string[] ErrorNodeMessages(IRootNode node);

    void AssertNoErrors();
*/

    setCurrentNode(node: IRootNode): void;
    reset(): void;
}

File: src/Lexy.Compiler/Parser/IParserLogger.cs
using System.Collections.Generic;
using Lexy.Compiler.Language;

namespace Lexy.Compiler.Parser;

public interface IParserLogger
{
    void LogInfo(string message);

    void Log(SourceReference reference, string message);
    void Fail(SourceReference reference, string message);

    bool HasErrors();
    bool HasComponentErrors();

    bool HasErrorMessage(string expectedError);

    public string FormatMessages();

    bool NodeHasErrors(IComponentNode node);

    string[] ErrorMessages();
    string[] ErrorComponentMessages();
    string[] ErrorNodeMessages(IComponentNode node);
    string[] ErrorNodesMessages(IEnumerable<IComponentNode> nodes);

    void AssertNoErrors();

    void SetCurrentNode(IComponentNode node);
    void ResetCurrentNode();
}